sympy/physics/mechanics/system.py

```diff
@@ -0,0 +1,441 @@
+    p : constants
+    q : generalized coordinates
+    u : generalized speeds
+    M : mass matrix or generalized inertia matrix (implicit dynamical equations
+    or combined dynamics and kinematics)
+    F : right hand side (implicit dynamical equations or combined dynamics and
+    kinematics)
+    G : right hand side of the kinematical differential equations
+
+
+
+    Attributes
+    ==========
+
+    coordinates : Matrix, shape(n, 1)
+        This is a matrix containing the generalized coordinates of the system
+
+    speeds : Matrix, shape(m, 1)
+        This is a matrix containing the generalized speeds of the system
+
+    states : Matrix, shape(o, 1)
+        This is a matrix containing the state variables of the system
+
+    alg_con : List
+        This list contains the indices of the algebraic constraints in the
+        combined equations of motion. The presence of these constraints requires
+        that a DAE solver be used instead of an ODE solver. If the system is
+        given in form [3] the alg_con variable will be adjusted such that it is
+        a representation of the combined kinematics and dynamics thus make sure
+        it always matches the mass matrix entered.
+
+    dyn_implicit_mat : Matrix, shape(m, m)
+        This is the M matrix in form [3] of the equations of motion (the mass
+        matrix or generalized inertia matrix of the dynamical equations of
+        motion in implicit form).
+
+    dyn_implicit_rhs : Matrix, shape(m, 1)
+        This is the F vector in form [3] of the equations of motion (the right
+        hand side of the dynamical equations of motion in implicit form).
+
+    comb_implicit_mat : Matrix, shape(o, o)
+        This is the M matrix in form [2] of the equations of motion. This matrix
+        contains a block diagonal structure where the top left block (the first
+        rows) represent the matrix in the implicit form of the kinematical
+        equations and the bottom right block (the last rows) represent the
+        matrix in the implicit form of the dynamical equations.
+
+    comb_implicit_rhs : Matrix, shape(o, 1)
+        This is the F vector in form [2] of the equations of motion. The top
+        part of the vector represents the right hand side of the implicit form
+        of the kinemaical equations and the bottom of the vector represents the
+        right hand side of the implicit form of the dynamical equations of
+        motion.
+
+    comb_explicit_rhs : Matrix, shape(o, 1)
+        This vector represents the right hand side of the combined equations of
+        motion in explicit form (form [1] from above).
+
+    kin_explicit_rhs : Matrix, shape(m, 1)
+        This is the right hand side of the explicit form of the kinematical
+        equations of motion as can be seen in form [3] (the G matrix).
+
+    dynamic_symbols : Matrix
+        This matrix contains all of the symbols that have a time dependence in
+        the equations of motion of the system
```

Comment:
Mention that these will have both the state variables, lagrange multipliers, and exogenous inputs.